### PR TITLE
recipes-robot/system-server: add opentrons-system-server recipe

### PR DESCRIPTION
**Pull request by fsinapi**
This PR adds the `opentrons-system-server` project to the OT3 image.

Changelog:
- Added a recipe for PyJWT version 2.6.0 (latest) and specified it as preferred, because the default in meta-python is 1.7.1
- Added a recipe for opentrons-system-server with a systemd service to run it as a daemon

Test steps:
- [x] Confirm build completes
- [x] Confirm that system-server is included in the version file
- [x] Upload built image to an OT-3
- [x] Confirm that `/opt/opentrons-system-server` exists on the filesystem + it has the expected contents
- [x] Confirm that `systemctl status opentrons-system-server` shows the system server as running
- [x] Run Python3 repl in opentrons-system-server and confirm that the PyJWT library exists + you can encode/decode using `HS512`

### CODE REVIEW

File: layers/meta-opentrons/recipes-robot/opentrons-system-server/opentrons-system-server.bb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    python3 ${S}/scripts/python_build_utils.py system-server ot3 dump_br_version > ${D}/opentrons_versions/opentrons-system-server-version.json
+
+    install -d ${D}/${systemd_system_unitdir}
+    install -m 0644 ${WORKDIR}/opentrons-system-server.service ${D}/${systemd_system_unitdir}/opentrons-system-server.service
+}
+
+FILES_${PN}_append = " ${systemd_system_unitdir/opentrons-system-server.service.d \
+                       ${systemd_system_unitdir}/opentrons-system-server.service.d/system-server-version.conf \
+                       ${sysconfdir}/udev/rules.d/95-opentrons-modules.rules \
+                       ${sysconfdir}/release-notes.md \
```
Comment (vegano1) [inline layers/meta-opentrons/recipes-robot/opentrons-system-server/opentrons-system-server.bb:41]: dont think we need these.

File: layers/meta-opentrons/recipes-robot/opentrons-system-server/files/opentrons-system-server.service [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Opentrons System HTTP Server
+Requires=nginx.service
+After=nginx.service
+
+[Service]
+Type=simple
+ExecStart=python3 -m system_server
+StateDirectory=opentrons-system-server
+Environment=PYTHONPATH=/opt/opentrons-system-server:/usr/lib/python3.8/site-packages\
```
Comment (sfoster1) [inline layers/meta-opentrons/recipes-robot/opentrons-system-server/files/opentrons-system-server.service:10]: what's that last backslash for?

File: layers/meta-opentrons/recipes-robot/opentrons-system-server/files/opentrons-system-server.service [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Opentrons System HTTP Server
+Requires=nginx.service
+After=nginx.service
+
+[Service]
+Type=simple
```
Comment (sfoster1) [inline layers/meta-opentrons/recipes-robot/opentrons-system-server/files/opentrons-system-server.service:7]: imo we should use `notify` and pysystemd `sd_notify` in the app

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Couple small things, see inline comments

File: layers/meta-opentrons/recipes-robot/opentrons-system-server/opentrons-system-server.bb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    install -d ${D}/${systemd_system_unitdir}
+    install -m 0644 ${WORKDIR}/opentrons-system-server.service ${D}/${systemd_system_unitdir}/opentrons-system-server.service
+}
+
+FILES_${PN}_append = " ${systemd_system_unitdir/opentrons-system-server.service.d \
+                       ${systemd_system_unitdir}/opentrons-system-server.service.d/system-server-version.conf \
+                       "
+
+RDEPENDS_${PN} += " python3-pyjwt nginx python-aionotify"
```
Comment (sfoster1) [inline layers/meta-opentrons/recipes-robot/opentrons-system-server/opentrons-system-server.bb:42]: what are we using aionotify for?

File: layers/meta-opentrons/recipes-robot/opentrons-system-server/files/opentrons-system-server.service (hunk 2 above)
Comment (fsinapi) [reply layers/meta-opentrons/recipes-robot/opentrons-system-server/files/opentrons-system-server.service:10]: typo

File: layers/meta-opentrons/recipes-robot/opentrons-system-server/opentrons-system-server.bb (hunk 4 above)
Comment (fsinapi) [reply layers/meta-opentrons/recipes-robot/opentrons-system-server/opentrons-system-server.bb:42]: accidental holdover

**Review by sfoster1 — APPROVED** [review]
Okay, looks good to me!

**Comment by vegano1** [comment]
lgtm!

**Review by vegano1 — APPROVED** [review]

**Comment by fsinapi** [comment]
Tested on robot after updating to `notify`, confirmed that the system-server starts up and notifies systemd that it is running (status is active)

**Pull request merged**